sycl/test/basic_tests/reduction/known_identities.cpp

```diff
@@ -0,0 +1,120 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %t.out
+
+// Tests the existance and specializations of known identities.
+
+#include <sycl/sycl.hpp>
+
+#include <cassert>
+#include <limits>
+#include <type_traits>
+
+template <typename BinOp, typename OperandT> constexpr void checkNoIdentity() {
+  static_assert(!sycl::has_known_identity<BinOp, OperandT>::value,
+                "Operation should not have a known identity!");
+  static_assert(!sycl::has_known_identity_v<BinOp, OperandT>,
+                "Operation should not have a known identity!");
+}
+
+template <typename BinOp, typename OperandT>
+void checkIdentity(OperandT Expected) {
+  static_assert(sycl::has_known_identity<BinOp, OperandT>::value,
+                "No trait specialization for known identity!");
+  static_assert(sycl::has_known_identity_v<BinOp, OperandT>,
+                "No trait specialization for known identity!");
+  assert((sycl::known_identity<BinOp, OperandT>::value == Expected) &&
+         "Identity does not match expected.");
+  assert((sycl::known_identity_v<BinOp, OperandT> == Expected) &&
+         "Identity does not match expected.");
```

Comment:
nit: Make function `constexpr` and move the `static_assert` to the caller to avoid the need to run the executable?

Alternatively, move the value to the template value param.

Reply:
I was unable to add the expected value as a template parameter, but I was able to make it compile-time-only using a macro.